# Eksploracyjna analiza danych i klasteryzacja 

## 1. Analiza danych 

### 1.1 Wczytanie danych bazowych

In [1]:
import pandas as pd
original_data = pd.read_json("cocktail_dataset.json")

In [2]:
original_data

id                   name        category                glass  \
0    11000                 Mojito        Cocktail       Highball glass   
1    11001          Old Fashioned        Cocktail  Old-fashioned glass   
2    11002        Long Island Tea  Ordinary Drink       Highball glass   
3    11003                Negroni  Ordinary Drink  Old-fashioned glass   
4    11004           Whiskey Sour  Ordinary Drink  Old-fashioned glass   
..     ...                    ...             ...                  ...   
129  11985  Quarter Deck Cocktail  Ordinary Drink       Cocktail glass   
130  11987              Queen Bee  Ordinary Drink       Cocktail glass   
131  11989        Queen Charlotte  Ordinary Drink        Collins glass   
132  11991        Queen Elizabeth  Ordinary Drink       Cocktail glass   
133  11993                Quentin  Ordinary Drink       Cocktail glass   

                                                  tags  \
0    [IBA, ContemporaryClassic, Alcoholic, USA, Asi...   
1         [IBA, Classic, Alcoholic, Expensive, Savory]   
2    [Strong, Asia, StrongFlavor, Brunch, Vegetaria...   
3                                       [IBA, Classic]   
4       [IBA, Classic, Alcoholic, ContemporaryClassic]   
..                                                 ...   
129                                               None   
130                                               None   
131                                               None   
132                                               None   
133                                               None   

                                          instructions  \
0    Muddle mint leaves with sugar and lime juice. ...   
1    Place sugar cube in old fashioned glass and sa...   
2    Combine all ingredients (except cola) and pour...   
3         Stir into glass over ice, garnish and serve.   
4    Shake with ice. Strain into chilled glass, gar...   
..                                                 ...   
129  Stir all ingredients with ice, strain into a c...   
130  Shake all ingredients with ice, strain into a ...   
131  Pour red wine and grenadine into a collins gla...   
132  Stir all ingredients with ice, strain into a c...   
133  In a shaker half-filled with ice cubes, combin...   

                                              imageUrl  alcoholic  \
0    https://cocktails.solvro.pl/images/ingredients...          1   
1    https://cocktails.solvro.pl/images/ingredients...          1   
2    https://cocktails.solvro.pl/images/ingredients...          1   
3    https://cocktails.solvro.pl/images/ingredients...          1   
4    https://cocktails.solvro.pl/images/ingredients...          1   
..                                                 ...        ...   
129  https://cocktails.solvro.pl/images/ingredients...          1   
130  https://cocktails.solvro.pl/images/ingredients...          1   
131  https://cocktails.solvro.pl/images/ingredients...          1   
132  https://cocktails.solvro.pl/images/ingredients...          1   
133  https://cocktails.solvro.pl/images/ingredients...          1   

                         createdAt                      updatedAt  \
0    2024-08-18T19:01:17.000+00:00  2024-08-18T19:06:16.000+00:00   
1    2024-08-18T19:01:58.000+00:00  2024-08-18T19:06:17.000+00:00   
2    2024-08-18T19:01:58.000+00:00  2024-08-18T19:06:17.000+00:00   
3    2024-08-18T19:01:58.000+00:00  2024-08-18T19:06:17.000+00:00   
4    2024-08-18T19:01:59.000+00:00  2024-08-18T19:06:18.000+00:00   
..                             ...                            ...   
129  2024-08-18T19:11:43.000+00:00  2024-08-18T19:11:43.000+00:00   
130  2024-08-18T19:11:44.000+00:00  2024-08-18T19:11:44.000+00:00   
131  2024-08-18T19:11:45.000+00:00  2024-08-18T19:11:45.000+00:00   
132  2024-08-18T19:11:45.000+00:00  2024-08-18T19:11:45.000+00:00   
133  2024-08-18T19:11:46.000+00:00  2024-08-18T19:11:46.000+00:00   

                                           ingredients  
0    [{'id

### 1.2 Wizualna analiza zbioru 

Patrząc na zbiór, można odrzucić takie kolumny jak :
- id
- imageUrl
- createdAt
- updatedAt	

Ze względu na brak posiadania jakichkolwiek informacji o elementach zbioru

### 1.3 Analiza wartości null oraz NaN

In [15]:
original_data.isna().value_counts()

id     name   category  glass  tags   instructions  imageUrl  alcoholic  createdAt  updatedAt  ingredients
False  False  False     False  True   False         False     False      False      False      False          99
                               False  False         False     False      False      False      False          35
Name: count, dtype: int64

In [31]:
original_data[original_data["tags"].isna()].count()

id              99
name            99
category        99
glass           99
tags             0
instructions    99
imageUrl        99
alcoholic       99
createdAt       99
updatedAt       99
ingredients     99
dtype: int64

Po wstępnej analizie - na 133 próbek 99 nie posiada tagów, co wynosi ok. 74,4% - można śmiało odrzucić tą kolumnę, ze względu na to, że dotyczy 1/4 zbioru danych.

In [ ]:
from matplotlib.pyplot import hist

hist(len(original_data["ingredients"]))
